### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by zschira**
Grouped together several small bug fixes in this PR.

### CODE REVIEW

File: src/pudl_archiver/archivers/eia860m.py [hunk 1]
```diff
@@ -31,13 +31,13 @@ async def get_resources(self) -> ArchiveAwaitable:
...
         self, link: str, match: typing.Match
     ) -> tuple[Path, dict]:
-        """Download zip file."""
+        """Download xlsx file."""
         url = f"https://eia.gov/{link}"
         year = match.group(2)
         month = self.month_map[match.group(1)]
-        download_path = self.download_directory / f"eia860m-{year}-{month}.zip"
-        await self.download_zipfile(url, download_path)
+        download_path = self.download_directory / f"eia860m-{year}-{month:02}.xlsx"
```
Comment (zschira) [inline src/pudl_archiver/archivers/eia860m.py:38]: Was naming files incorrectly, and setting parts wrong.

File: src/pudl_archiver/zenodo/entities.py [hunk 2]
```diff
@@ -13,13 +14,13 @@
 class Doi(ConstrainedStr):
```
Comment (zschira) [inline src/pudl_archiver/zenodo/entities.py:14]: For some reason, when creating `DatasetSettings` objects I was getting validation errors. Making this explicitly compiled regex patterns fixed the problem for me. Also added unit tests to test this.

File: src/pudl_archiver/zenodo/entities.py [hunk 3]
```diff
@@ -81,6 +82,17 @@ def check_empty_string(cls, doi: str):  # noqa: N805
     def from_data_source(cls, data_source_id: str) -> "DepositionMetadata":
         """Construct deposition metadata object from PUDL DataSource model."""
         data_source = DataSource.from_id(data_source_id)
+        creators = [
```
Comment (zschira) [inline src/pudl_archiver/zenodo/entities.py:85]: Some new data sources don't have contributors defined in PUDL

File: pyproject.toml [hunk 4]
```diff
@@ -9,7 +9,7 @@ dependencies = [
     "coloredlogs~=15.0",
     "feedparser>=6.0",
     "tqdm>=4.64",
-    "catalystcoop.arelle-mirror==1.3.0",
+    "arelle-release>=2.3,<2.4",
```
Comment (jdangerx) [inline pyproject.toml:12]: woo!

File: src/pudl_archiver/archivers/eia860m.py (hunk 1 above)
Comment (jdangerx) [reply src/pudl_archiver/archivers/eia860m.py:38]: Nice find :)

File: src/pudl_archiver/zenodo/entities.py (hunk 3 above)
Comment (jdangerx) [reply src/pudl_archiver/zenodo/entities.py:85]: Makes sense, and then we can just claim credit :)

File: tests/unit/zenodo_api_test.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+"""Test zenodo API client."""
+import pydantic
+import pytest
+
+from pudl_archiver.zenodo.api_client import DatasetSettings
+
+
+def test_dataset_settings():
```
Comment (jdangerx) [inline tests/unit/zenodo_api_test.py:8]: non-blocking: These tests seem like they're mostly testing if Pydantic works the way we expect it to - which is fine, but we might not need these? Separately, I think we might want some "test to see if our dataset settings updating logic does the right thing" tests, though those seem a bit more annoying to write.

**Review by jdangerx — APPROVED** [review]
🎉

File: tests/unit/zenodo_api_test.py (hunk 5 above)
Comment (zschira) [reply tests/unit/zenodo_api_test.py:8]: My thoughts were that these sort of test that the regex is working as expected by saying "give it something that looks like a DOI and it's all good. Give it anything else and it fails". Tests that actually test that the settings are handled/updated properly would definitely be more useful though.

**Pull request merged**